examples/morpho_blue.ipynb

```diff
@@ -0,0 +1,1018 @@
+    "\n",
+    "BLUE_CONTRACT_ADDRESS = \"0xBBBBBbbBBb9cC5e90e3b3Af64bdAF62C37EEFFCb\"\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Utils"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "\n",
+    "def convert_to_type(x, type, name=\"\", data_offset = 0):\n",
+    "    ''' Convert raw data to a specific type\n",
+    "    x: the raw data (bytes)\n",
+    "    type: the type of the data\n",
+    "    name: the name of the data\n",
+    "    data_offset: the offset of the data in the raw data'''\n",
+    "    \n",
+    "    if type == 'string':\n",
+    "        offset = int(x[data_offset+12: data_offset + 32].hex(), 16)\n",
+    "        length = int(x[offset:offset + 32].hex(), 16)\n",
+    "        return x[offset + 32: offset + 32 + length].decode('utf-8')\n",
+    "    elif '[]' in type:\n",
+    "        # TODO\n",
+    "        return '0x'\n",
+    "    elif len(type) > 4 and type[:4] == 'uint':\n",
+    "        return float(int(x[data_offset+22: data_offset+32].hex(), 16))\n",
+    "    elif type == 'address':\n",
+    "        return '0x' + x[12+data_offset:32+data_offset].hex()\n",
+    "    elif type == 'bool':\n",
+    "        return int(x[data_offset: 32 + data_offset], 16) > 0\n",
+    "    elif len(type) > 5 and type[:5] == 'bytes':\n",
+    "        bytes_left = 32 - int(type[5:])\n",
+    "        return '0x' + x[data_offset+bytes_left: data_offset+32].hex()\n",
+    "    else:\n",
+    "        return x\n",
+    "    \n",
+    "def parse_raw_events(raw_events: pl.DataFrame, signature: str) -> pl.DataFrame:\n",
+    "    '''\n",
+    "    Parse raw events from a DataFrame, this dataframe should contain the following columns:\n",
+    "    - topic0 to topic3: the topics of the event\n",
+    "    - data: the data of the event\n",
+    "    The signature is used to determine the type of the event, it should look like this:\n",
+    "    \"EventName(uint256 indexed arg1, uint256 indexed arg2, T arg3, ...)\"\n",
+    "\n",
+    "    The function works with basic types (string, uint, address, bool, bytes) but not yet (TODO) with arrays or structs.\n",
+    "    '''\n",
+    "    \n",
+    "    # Parse the signature\n",
+    "    first_parenthesis = signature.find('(')\n",
+    "    last_parenthesis = signature.rfind(')')\n",
+    "\n",
+    "    print(f'Parsing Event name: {signature[:first_parenthesis]}')\n",
+    "\n",
+    "    types_str = signature[first_parenthesis + 1:last_parenthesis]\n",
+    "    types = types_str.split(',')\n",
+    "    indexed_types, data_types = [], []\n",
+    "\n",
+    "    for t in types:\n",
+    "        parts = t.strip().split(' ')\n",
+    "        if 'indexed' in parts:\n",
+    "            indexed_types.append((parts[0], parts[-1]))\n",
+    "        else:\n",
+    "            data_types.append((parts[0], parts[-1]))\n",
```

Comment:
it will break when the name is not specified (and I think that it's optionnal)

Reply:
it's not optional in the signature but it's quite cool to have the column renamed to the attribute name